src/cmd/services/m3coordinator/downsample/downsampler_test.go

```diff
@@ -2841,6 +2842,53 @@ func TestDownsamplerWithOverrideNamespace(t *testing.T) {
 	testDownsamplerAggregation(t, testDownsampler)
 }
 
+func TestSafeguardInProcessDownsampler(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	store := kv.NewMockStore(ctrl)
+	store.EXPECT().SetIfNotExists(gomock.Eq(matcher.NewOptions().NamespacesKey()), gomock.Any()).Return(0, nil).Times(1)
+	store.EXPECT().Set(gomock.Any(), gomock.Any()).Times(0)
+	store.EXPECT().CheckAndSet(gomock.Any(), gomock.Any(), gomock.Any()).Times(0)
+	store.EXPECT().Delete(gomock.Any()).Times(0)
+	_ = newTestDownsampler(t, testDownsamplerOptions{
+		remoteClientMock: nil,
+		kvStore:          store,
+	})
+}
+
+func TestDownsamplerNamespacesEtcdInit(t *testing.T) {
+	t.Run("does not reset namespaces key", func(t *testing.T) {
```

Comment:
Wouldn't it be cleaner to have separate test methods instead of all those `t.Run`? Or was your goal to have nice test descriptions? I wish `testing.T` simply had some kind of `SetName` for this.

Reply:
I was trying to group those namespace init related tests. This test file is 3k lines long. So different test methods can just get lost.